src/command/process_update_task.rs

```diff
@@ -639,15 +697,37 @@ impl AsyncRunnable for ProcessUpdateTask {
             }
         };
 
-        if let Err(error) = processor.process().await {
-            log::error!(
-                "Failed to process the update {:?} - {:?}. Reverting...",
-                self.update,
-                error
-            );
+        match processor.process().await {
+            Err(error) => {
+                log::error!(
+                    "Failed to process the update {:?} - {:?}. Reverting...",
+                    self.update,
+                    error
+                );
 
-            if let Err(err) = processor.revert_state().await {
-                log::error!("Failed to revert: {:?}", err);
+                if let Err(err) = processor.revert_state().await {
+                    log::error!("Failed to revert: {:?}", err);
+                }
+            }
+
+            Ok(option) => {
+                if let Some(vec) = option {
+                    let tasks: Vec<ScheduleWeatherTask> = vec
+                        .into_iter()
+                        .map(|forecast| {
+                            ScheduleWeatherTask::builder()
+                                .cron_expression(forecast.cron_expression)
+                                .chat_id(forecast.chat_id)
+                                .user_id(forecast.user_id)
+                                .city_id(forecast.city_id)
+                                .build()
+                        })
+                        .collect();
+
+                    for task in tasks {
```

Comment:
1. I think you're removing  old tasks but you're not creating  a new task
2. Maybe you can remove all tasks from the user user?

Because currently the flow of adding and maintaining multiple forecasts is very messy

Reply:
This code it is just for remove the task, i need to build the tasks because i need to delete it from Fang tables.
Maybe we can advice the user that maybe he gets a last update idk.

Reply:
discuss #34